temporal-testing-junit4/src/main/java/io/temporal/testing/TestWorkflowRule.java

```diff
@@ -84,14 +85,16 @@ private TestWorkflowRule(
       Object[] activityImplementations,
       WorkerOptions workerOptions,
       long testTimeoutSeconds,
-      boolean doNotStart) {
+      boolean doNotStart,
+      TracingWorkerInterceptor tracer) {
```

Comment:
Instead of taking specific implementation, let's have a list or var-arg array of WorkerInterceptor here. This would be more generic and will allow users to use any custom interceptors when needed.